test/library/encryption/assertion.go

```
@@ -390,3 +394,72 @@ func assertWellKnownRoutes(t testing.TB, etcdClient EtcdClient, expectedMode str
+// assertKMSPreflightSucceeded asserts preflight passed for the CR's current config: degraded is
+// False, preflight reports Succeeded for the observed config hash, remoteKeyID is set (proving a
+// live KMS check ran), and remoteKeyID advanced when the config changed since previous.
+func assertKMSPreflightSucceeded(ctx context.Context, t testing.TB, dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, name string, previous operatorv1.KMSPreflightCheck) {
+	t.Helper()
+
+	var preflight operatorv1.KMSPreflightCheck
+	var degradedFalse bool
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, time.Minute, true, func(ctx context.Context) (bool, error) {
+		obj, err := dynamicClient.Resource(gvr).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 445) but it returns nil

(nilerr)